### ISSUE TITLE
Package name redeclared when shadowing packages and using pkg flag

### ISSUE THREAD

**Issue by joncalhoun** [matryer/moq#94]
This probably doesn't affect many people, and it isn't super important, but let's say you create a package with a name that is the same as a package it uses. Eg I might make `github.com/xxx/myapp/http` and inside of it use the `net/http` package.

```go
package http

//go:generate moq -pkg http_test -out thing_moq_test.go . Thing

type Thing interface {
  Blah(w http.ResponseWriter, r *http.Request)
}
```

If I do this AND use the `-pkg` flag, the following is generated:

```go
import (
  "github.com/xxx/myapp/http"
  "net/http"
  "sync"
)
```

And this doesn't work 😢 

This isn't urgent at all as I can just NOT use the `-pkg` flag and make it work. I'm just pointing it out because I ran into it today.

**Comment by omarkohl** [comment]
This _could_ be related to #92 . If you want you could try pull request #93 to verify if it fixes the issue.

**Comment by joncalhoun** [comment]
Does #93 introduce named imports? Because that is really the only fix here. One of the two `http` packages needs to be named. Eg:

```go
import (
  myapp_http "github.com/xxx/myapp/http"
  "net/http"
  "sync"
)
```

Including both imports isn't actually a bug; it is necessary for the code to work.

**Comment by omarkohl** [comment]
You are right both imports are required. #92 happens because an unnecessary import is made. So probably it's not related.

**Comment by hexdigest** [comment]
@joncalhoun @omarkohl I know it may look like advertisement and it surely is :) but take a look at [minimock](https://github.com/gojuno/minimock) Most of the issues you guys create here either never existed in minimock or solved long time ago.

**Comment by gonzaloserrano** [comment]
Same thing happening here :-)

**Comment by kl09** [comment]
This error can be resolved by fmt=goimports

**Comment by sudo-suhas** [comment]
Closed via #141.
